core/src/system_monitor_service.rs

```diff
@@ -728,25 +740,26 @@ impl SystemMonitorService {
                 edx: 0,
             };
 
-            let max_leaf = cpuid_mfr.eax;
-            let max_subleaf_7 = if 7 <= max_leaf {
-                __get_cpuid_max(7).1
-            } else {
-                0
+            let max_leaf = match CpuidParamValue::try_from(std::cmp::min(
+                cpuid_mfr.eax,
+                CPUID_PARAM_MAX_SUPPORTED_VALUE,
+            )) {
+                Ok(val) => val,
+                Err(_err) => CpuidParamValue::Manufacturer,
             };
 
             let mfr_id: i64 = if cpuid_mfr.ebx == CPUID_MANUFACTURER_EBX_INTEL
                 && cpuid_mfr.edx == CPUID_MANUFACTURER_EDX_INTEL
                 && cpuid_mfr.ecx == CPUID_MANUFACTURER_ECX_INTEL
             {
-                CpuManufacturer::Intel // GenuineIntel
+                CpuManufacturer::Intel.into() // GenuineIntel
             } else if cpuid_mfr.ebx == CPUID_MANUFACTURER_EBX_AMD
                 && cpuid_mfr.edx == CPUID_MANUFACTURER_EDX_AMD
                 && cpuid_mfr.ecx == CPUID_MANUFACTURER_ECX_AMD
             {
-                CpuManufacturer::AMD // AuthenticAMD
+                CpuManufacturer::Amd.into() // AuthenticAMD
             } else {
-                CpuManufacturer::Other // anything else
+                CpuManufacturer::Other.into() // anything else
             };
 
             let cpuid_processor = if CpuidParamValue::Processor <= max_leaf {
```

Comment:
prefer coercing the `CpuidParamValue`s into `u32` for the `__cpuid()` calls below